File: tmp_check_items.py
import urllib.request
import json

try:
    with urllib.request.urlopen('http://localhost:5000/api/items', timeout=10) as resp:
        data = resp.read().decode('utf-8')
        print('STATUS', resp.status)
        parsed = json.loads(data)
        print('TYPE', type(parsed).__name__)
        if isinstance(parsed, list):
            print('LENGTH', len(parsed))
        else:
            print(parsed)
except Exception as e:
    print('ERROR', repr(e))

File: tmp_check_items.js
const http = require('http');
const url = 'http://localhost:5000/api/items';

http.get(url, (res) => {
  let data = '';
  res.on('data', (chunk) => { data += chunk; });
  res.on('end', () => {
    console.log('STATUS', res.statusCode);
    try {
      const parsed = JSON.parse(data);
      console.log('TYPE', Array.isArray(parsed) ? 'array' : typeof parsed);
      if (Array.isArray(parsed)) console.log('LENGTH', parsed.length);
      else console.log(parsed);
    } catch (err) {
      console.error('PARSE ERROR', err.message);
      console.log(data);
    }
  });
}).on('error', (err) => {
  console.error('ERROR', err.message);
});
